src/index.ts

```diff
@@ -246,7 +246,14 @@ class PluginPlayground {
 
     // Unregister plugin if already registered.
     if (this.app.hasPlugin(plugin.id)) {
-      delete (this.app as any)._pluginMap[plugin.id];
+      try {
+        if (plugin.deactivate) {
+          await this.app.deactivatePlugin(plugin.id);
+        }
+        this.app.deregisterPlugin(plugin.id, true);
+      } catch (reason) {
+        showErrorMessage('Deactivation of the old plugin failed', reason);
```

Comment:
We should probably preserve compatibility with 3.x branch here.  Below is a suggestion for one way to do so (though checking for `this.app.deactivatePlugin`/`this.app.deregisterPlugin` could be better).

```suggestion
        try {
          if (plugin.deactivate) {
            plugin.deactivate()
          }
          delete (this.app as any)._pluginMap[plugin.id];
        } catch (otherReason) {
          showErrorMessage('Deactivation of the old plugin failed', reason, otherReason);
        }
```